src/experimentExtractor/tests/webpack-plugin.spec.ts

```diff
@@ -0,0 +1,166 @@
+import assert from "assert";
+import { beforeEach, describe, it } from "mocha";
+import path from "path";
+import webpack from "webpack";
+import ExtractExperimentsPlugin from "../index";
+import fs from "fs-extra";
+
+describe("experimentExtractor/webpack-plugin", function () {
+  const resultFilePath = path.resolve(__dirname, "tmp", "experiments.json");
+
+  const getWebpackConfig = (entryFilePath, entryName = "index") => {
+    const entry = {};
+    entry[entryName] = path.resolve(
+      __dirname,
+      "webpackFiles",
+      ...entryFilePath
+    );
+
+    return {
+      target: "node",
+      mode: "production",
+      entry: entry,
+      resolve: {
+        extensions: [".jsx", ".tsx", ".ts", ".js"],
+      },
+      output: {
+        libraryTarget: "commonjs",
+        filename: "[name]/index.js",
+        path: path.resolve(__dirname, "tmp"),
+      },
+      module: {
+        rules: [
+          {
+            test: /\.[jt]sx?/,
+            include: path.resolve(__dirname, "..", ".."),
+            exclude: /node_modules/,
+            use: {
+              loader: "ts-loader",
+              options: { reportFiles: ["src/**/*.{ts,tsx}", "!**.spec.ts"] },
+            },
+          },
+        ],
+      },
+      plugins: [
+        new ExtractExperimentsPlugin({
+          resultFilePath: resultFilePath,
+          isDebug: false,
+          libraryName: "abTestingInfra",
+        }),
+      ],
+    };
+  };
+
+  const runWebpack = (entryFilePath, entryName = undefined) =>
+    new Promise((resolve, reject) => {
+      webpack(getWebpackConfig(entryFilePath, entryName), (err, stats) => {
+        if (err) {
+          reject(err);
+        } else if (stats.hasErrors()) {
+          reject(stats.toString());
+        }
+        resolve();
+      });
+    });
```

Comment:
Do these need to be defined locally in the `describe` block? Else I suggest we pull them out to the top-level of the file

Reply:
Done!